Blank Canvas: Add block patterns

kjellr opened the pull request:
This PR adds a set of single-page site block patterns to the Blank Canvas theme. 

Currently included: 

About Me|Invitation
---|---
![Screen Shot 2021-01-12 at 11 25 53 AM](https://user-images.githubusercontent.com/1202812/104354788-bc5f5c00-54d7-11eb-9cce-3819fc2704de.png)|![Screen Shot 2021-01-12 at 1 11 28 PM](https://user-images.githubusercontent.com/1202812/104354790-bcf7f280-54d7-11eb-8380-f7b0071baa27.png)

Links|Card
---|---
![Screen Shot 2021-01-12 at 1 09 58 PM](https://user-images.githubusercontent.com/1202812/104354791-bcf7f280-54d7-11eb-8d34-e24471142e67.png)|![Screen Shot 2021-01-12 at 1 09 13 PM](https://user-images.githubusercontent.com/1202812/104354792-bd908900-54d7-11eb-9275-878a4937bc8e.png)

These still need some tidying up style-wise.

kjellr committed:
Add About Me + Links Patterns

kjellr committed:
Add invitation pattern.

kjellr committed:
Add a "card" pattern.

ianstewart commented:
The the alignment of the photo subject and the buttons seems to clash a bit with the left alignment  of the heading and text in the card pattern. I wonder if playing around with a different image, cropping or alignment there could brighten it up? The photo also a bit of an apocalyptic look with the pattern being overall more "dark" in contrast with the other patterns.

Perhaps the invitation pattern could be have the text or overall height reduced a bit? The two, tall rectangles next to each other just feels a bit static or not as dynamic as it perhaps could be?

It might be nice for "theme cohesion" if the photos in the patterns could all get the same "feel" but this may be difficult if we're looking to use GPL-compatible images. Along the same vein, did you try out using the same heading weights?

ianstewart commented:
Looks like a right single quote wound up missing in this one …

![image](https://user-images.githubusercontent.com/545779/104376708-0140be00-54eb-11eb-82e0-9e1722b3b9a4.png)

melchoyce commented:
Let's update that copy entirely:

> You're Invited
> 
> Pier 38
> San Francisco, CA
> 
> 8pm–12am
> February 24th
> 
> Join us for a celebratory party featuring creative cocktails and mocktails, seasonal chef-crafted snacks and appetizers, and a range of desserts and pastries from local bakeries. 

![image](https://user-images.githubusercontent.com/2846578/104380986-d873f500-54f9-11eb-87fd-c523d2f3b3fa.png)

kjellr committed:
Revise patterns.

kjellr committed:
Adjust responsive width of alignwide.

kjellr committed:
Add a spacer to the About Me pattern.

kjellr commented:
Here are a few updates. I've:

- Adjusted the text for the invitation pattern.
- Synced these up better from a colors + photos perspective. 
- Added a full-screen "Split" pattern. 
- Moved the "Card" one to be WP.com-only, since I couldn't get it to work nicely without the layout grid block.
- Widened the wide alignment width so it's more clearly differentiated from the standard width.

![Screen Shot 2021-01-13 at 12 16 28 PM](https://user-images.githubusercontent.com/1202812/104486503-da8d9080-5599-11eb-8f3f-4b5c8803a1f3.png)
![Screen Shot 2021-01-13 at 12 16 34 PM](https://user-images.githubusercontent.com/1202812/104486505-db262700-5599-11eb-8764-6be44b2d2705.png)
![Screen Shot 2021-01-13 at 12 16 48 PM](https://user-images.githubusercontent.com/1202812/104486506-db262700-5599-11eb-8f01-4ad238f19100.png)
![Screen Shot 2021-01-13 at 12 17 51 PM](https://user-images.githubusercontent.com/1202812/104486507-db262700-5599-11eb-840a-beedf8af3c5d.png)
![Screen Shot 2021-01-13 at 11 55 20 AM](https://user-images.githubusercontent.com/1202812/104486512-dcefea80-5599-11eb-99cc-551adf0b4b54.png)

kjellr committed:
Merge branch 'trunk' into update/blank-canvas-patterns

kjellr committed:
Update image credits.

kjellr commented:
@melchoyce is also looking into adding one more pattern that would include a contact form. 👍

kjellr committed:
Add email signup pattern.

kjellr committed:
Copy Edit

kjellr commented:
The latest commit adds an email signup pattern: 

![Screen Shot 2021-01-14 at 9 34 50 AM](https://user-images.githubusercontent.com/1202812/104604818-c3a77680-564b-11eb-97b1-9404012f083f.png)

@melchoyce, I started with one of your layouts, but changed the topic a bit to fit a colorful photo like the other ones we're using here.

This seems like a decent set of patterns to start with — I think this should be ready for a review!

kjellr committed:
Adjust invitation pattern.

kjellr committed:
Merge branch 'trunk' into update/blank-canvas-patterns

scruffian commented:
This looks good. What would be amazing is if the patterns could live in separate files and then the function pulls from them, so that they are easier to maintain...

kjellr commented:
Are you envisioning a separate "block-patterns" directory, that would house each of these as individual files? That could be cool. I'd suggest we explore that in a followup PR though.

ianstewart commented on blank-canvas/inc/wpcom.php:
```diff
@@ -23,7 +23,7 @@ function blank_canvas_register_wpcom_block_patterns() {
 		if ( function_exists( 'register_block_pattern' ) ) {
 
-			/* This pattern is WP.com-Only Beacuse it relies on the Layout Grid block. */
+			/* This pattern is WP.com-only because it relies on the Layout Grid block. */
```
@kjellr is there some additional function_exists() to check if Jetpack Layout Grid is present that could be added here? Or are there any WordPress plugin or block check functions that could be checked here for throwing notices in /wp-admin?

kjellr replied:
I did look into that briefly, but I didn't find anything obvious. There don't seem to be any functions I could check for in [the plugin source](https://plugins.svn.wordpress.org/layout-grid/trunk/). 

That could use another set of eyes though if someone else has a moment.

kjellr replied:
So it _really_ looks like this should work: 

```
if ( class_exists( 'WP_Block_Type_Registry' ) && WP_Block_Type_Registry::get_instance()->is_registered( 'jetpack/layout-grid' ) ) {
    register_block_pattern ( ... )
}
```

But for some reason I can't get it to. 😩

kjellr replied:
@MaggieCabrera and I took a look at getting that working on Friday, but we didn't make any progress. 

Since the current solution (load these only on WP.com) technically works ok, I suggest we merge and follow up with a better fix later on. 

The only effected parties here are folks who use this theme outside of WP.com and also have the Layout Grid plugin installed. And even in that case, they'll currently just not see this pattern, which seems fine. 

Does that work for folks? 

If so, would someone mind giving this a final review?

kjellr committed:
Fix markup error for invitation pattern.

kjellr committed:
Fix typo

pbking commented:
I notice that for the "Links" Block Pattern the links alignment is different for .com and .org.
![image](https://user-images.githubusercontent.com/146530/105047966-03e66a80-5a39-11eb-9576-b686e3c3cc27.png)

kjellr commented:
@pbking did you have the Gutenberg plugin activated there? The pattern uses a 100% width control that is currently only available in the plugin (which is loaded by default on .com). 

If the plugin is turned off, things are left aligned and they wrap: 

![Screen Shot 2021-01-19 at 9 47 27 AM](https://user-images.githubusercontent.com/1202812/105050193-8708c000-5a3b-11eb-9a60-6a44a6715c1c.png)

I think that's ok for now though, since we're only launching on WP.com to start, and I don't think it makes sense to write custom code to fix something that'll work fine out of the box once this version of the plugin is merged into core.

kjellr committed:
Merge branch 'trunk' into update/blank-canvas-patterns

pbking commented:
Indeed, installing the Gutenberg plugin addressed the alignment issue.  If that's OK then it all LGTM.

kjellr commented:
Thanks!
